backend/canisters/identity/impl/src/queries/get_account_linking_code.rs

```diff
@@ -0,0 +1,33 @@
+use crate::{RuntimeState, read_state};
+use canister_api_macros::query;
+use identity_canister::get_account_linking_code::{Response::*, *};
+use types::{AccountLinkingCode, UserId};
+
+#[query(msgpack = true, candid = true)]
+fn get_account_linking_code(_args: Args) -> Response {
+    read_state(get_account_linking_code_impl)
+}
+
+// Fetches existing account linking code for the caller's user ID.
+fn get_account_linking_code_impl(state: &RuntimeState) -> Response {
+    let now = state.env.now();
+
+    if let Some(user_id) = state.get_user_id_by_caller() {
```

Comment:
You could remove the duplicate `NotFound` cases by returning in the `Success` case and then having `NotFound` as the fallback on the last line.